zebra/zebra_nhg.c

```diff
@@ -591,10 +589,11 @@ zebra_nhg_find_nexthop(uint32_t id, struct nexthop *nh, afi_t afi, int type)
 {
 	struct nhg_hash_entry *nhe = NULL;
 	struct nexthop_group nhg = {};
+	vrf_id_t vrf_id = !vrf_is_backend_netns() ? VRF_DEFAULT : nh->vrf_id;
```

Comment:
I'm having trouble following this: don't we have to include the nexthop's vrf in this lookup? I agree that it's awkward that sometimes there's a value in an nhe that's copied out of a single nexthop, and sometimes not.

Reply:
nexthop groups are namespace based *not* vrf based.  So if you are not using namespaced backends for a vrf you need to make this distinction.  Else you end up with a some scenarios where we are reporting a nexthop group in VRF X and that means nothing if you are using l3mdev's.  I would rather just say they are in the default vrf.